src/OAuth2/GrantType/Facebook.php

```diff
@@ -54,6 +54,12 @@ protected function getTokenInfo($token)
                 return null;
             }
 
+            // do not accept tokens generated not for our application even if they are valid,
+            // to protect against "man in the middle" attack
+            $tokenMetadata = $this->facebook->getOAuth2Client()->debugToken($token);
```

Comment:
this is only required with older versions of facebook sdk, but to be sure I will keep it because it can change at any time(wouldn't be the first time).